lib/analysis/author/community.py

```diff
@@ -0,0 +1,265 @@
+    author_graph = nx.DiGraph()
+    for msg_id, message in json_data.items():
+        if message['From'] in top_authors:
+            if message['Cc'] is None:
+                addr_list = message['To']
+            else:
+                addr_list = message['To'] | message['Cc']
+            for to_address in addr_list:
+                if to_address in top_authors:
+                    if author_graph.has_edge(message['From'], to_address):
+                        author_graph[message['From']][to_address]['weight'] *= \
+                            author_graph[message['From']][to_address]['weight'] / (author_graph[message['From']][to_address]['weight'] + 1)
+                    else:
+                        author_graph.add_edge(message['From'], to_address, weight=1)
+
+    author_graph_undirected = author_graph.to_undirected()
+    clustering_coeff = nx.clustering(author_graph_undirected)
+    in_degree_dict = author_graph.in_degree(nbunch=author_graph.nodes_iter())
+    out_degree_dict = author_graph.out_degree(nbunch=author_graph.nodes_iter())
+
+    for email_addr in top_authors:
+        top_authors_data[email_addr].append(in_degree_dict[email_addr])
+        top_authors_data[email_addr].append(out_degree_dict[email_addr])
+        top_authors_data[email_addr].append(clustering_coeff[email_addr])
+
+    print("Parsing", tree_filename + "...")
+    with open(tree_filename, 'r') as tree_file:
+        for line in tree_file:
+            if not line or line[0] == '#':
+                continue
+            line = line.split()
+            if line[2][1:-1] in top_authors:
+                top_authors_data[line[2][1:-1]].append(float(line[1]))
+        tree_file.close()
+
+    with open("top_authors_data.csv", 'w') as output_file:
+        output_file.write("Email Address,Author Score,In-Degree,Out-Degree,Clustering Coeff,Module Flow\n")
+        for email_addr, data_list in top_authors_data.items():
+            output_file.write(email_addr+","+",".join([str(x) for x in data_list])+"\n")
+        output_file.close()
+    print("Authors data written to file.")
+
+
+def write_pajek(author_graph, filename="author_graph.net"):
+    # Write Pajek file compatible with the Infomap Community Detection module
+    nx.write_pajek(author_graph, filename)
+    lines_in_file= list()
+    with open(filename, 'r') as pajek_file:
+        for line in pajek_file:
+            lines_in_file.append(line)
+    num_vertices = int(lines_in_file[0].split()[1])
+    for i in range(1, num_vertices+1):
+        line = lines_in_file[i].split()
+        line[1] = "\"" + line[1] + "\""
+        del line[2:]
+        line.append("\n")
+        lines_in_file[i] = " ".join(line)
+    with open(filename, 'w') as pajek_file:
+        for line in lines_in_file:
+            pajek_file.write(line)
+    print("Written to:", filename)
+
+
+def write_pajek_for_submodules(json_data, tree_filename="infomap/output/"+"author_graph.tree"):
```

Comment:
same comments as above. sphinx docs and potential candidate for moving to lib/util module.

Reply:
Please check b8193f4c7935c85e262dfae192c1b2d30abbec7d for documentation on community module. I have addressed the issue of moving to util folder above.